apps/userdashboard/assets/js/a4_candy_userdashboard/FilterBar.jsx

```diff
@@ -1,29 +1,49 @@
-import React from 'react'
+import React, { useState } from 'react'
 import django from 'django'
 
-export const FilterBar = () => {
+const getItemByFilter = (items, filter) => {
+  return items.find(item => item.filter === filter)
+}
+
+export const FilterBar = (props) => {
+  const filterItems = [
+    { name: 'Pending', filter: '?is_pending=true' },
+    { name: 'Archived', filter: '?is_pending=false' },
+    { name: 'All', filter: '' }
+  ]
+  const [currFilterItem, setCurrFilterItem] =
+    useState(getItemByFilter(filterItems, props.selectedFilter))
+
+  const onSelectFilter = (filterItem) => {
+    setCurrFilterItem(filterItem)
+    props.onFilterChange(filterItem.filter)
+  }
+
   return (
-    <div className="filter-bar justify-content-end" role="group" aria-label={django.gettext('Filter')}>
-      <div className="me-sm-1 mt-2 mt-sm-0">
+    <div
+      className="filter-bar justify-content-end"
+      role="group"
```

Comment:
I don't think we should use `group` without using `groupitem`? but it's only just something i've learned about so it's possible we do it wrong elsewhere https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_group_role but semantically I don't think it makes sense